rental.rb

```diff
@@ -0,0 +1,20 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
+
+  def book=(book)
+    @book = book
+    book.students.push(self) unless book.students.include?(self)
+  end
+
+  def person=(person)
+    @person = person
+    person.students.push(self) unless person.students.include?(self)
+  end
```

Comment:
- [ ] Now, in line with the above review thread. It would be better if we implemented the associations as required of us in the project requirements. Please take a look at this last bullet point in the project requirements:

---
- **_`Modify the constructor of Rental so Book and Person are set in it.`_**
---

The constructor here refers to the **_`initialize`_** method of this class. Consider refactoring the logic of these two classical setter methods back into the **_`initialize`_** method.

## Some correction
In both methods, you are pushing to **`students`** instead of **`rentals`**. Please correct this one too. It should be:

```ruby
book.rentals.push(self)
person.rentals.push(self)
```

All this refactoring will necessitate that all the class attributes be readable and writable.

Reply:
Thank you for all your suggestions! They are very useful and well explained. I only have one question: When I implement the logic of these two methods in the constructor, I have to leave those methods there or it would be better if I remove them?

Reply:
You are welcome @nachosala89. You'll have to remove them.

Reply:
Thank you again!